odoo/addons/base/models/res_company.py

```diff
@@ -158,6 +159,11 @@ def _compute_logo_web(self):
             img = company.partner_id.image_1920
             company.logo_web = img and base64.b64encode(tools.image_process(base64.b64decode(img), size=(180, 0)))
 
+    @api.depends('partner_id.image_1920', 'logo')
+    def _compute_uses_default_logo(self):
+        for company in self:
+            self.uses_default_logo = company.logo and company.logo == self._get_logo()
+
```

Comment:
Read the logo only once from disk
```suggestion
    @api.depends('partner_id.image_1920', 'logo')
    def _compute_uses_default_logo(self):
        default_logo = self._get_logo()
        for company in self:
            self.uses_default_logo = (company.logo == default_logo)

```

Also, since we're at it, can you modify the `_get_logo` function too? For that unclosed file descriptor...

```diff
diff --git a/odoo/addons/base/models/res_company.py b/odoo/addons/base/models/res_company.py
index 07828f82c6ef..b88ff733d48b 100644
--- a/odoo/addons/base/models/res_company.py
+++ b/odoo/addons/base/models/res_company.py
@@ -10,6 +10,7 @@ import warnings
 from odoo import api, fields, models, tools, _, Command
 from odoo.exceptions import ValidationError, UserError
 from odoo.modules.module import get_resource_path
+from odoo.tools import file_open
 from random import randrange
 from PIL import Image
 
@@ -25,7 +26,8 @@ class Company(models.Model):
         raise UserError(_('Duplicating a company is not allowed. Please create a new company instead.'))
 
     def _get_logo(self):
-        return base64.b64encode(open(os.path.join(tools.config['root_path'], 'addons', 'base', 'static', 'img', 'res_company_logo.png'), 'rb') .read())
+        with file_open('base/static/img/res_company_logo.png', 'rb') as file:
+            return base64.b64encode(file.read())
 
     def _default_currency_id(self):
         return self.env.user.company_id.currency_id
```

Reply:
good idea :)